src/Uno.UWP/UI/Input/GestureRecognizer.Manipulation.InertiaProcessor.cs

```diff
@@ -0,0 +1,190 @@
+					}
+					if (IsNaN(DesiredExpansionDeceleration))
+					{
+						DesiredExpansionDeceleration = .001;
+					}
+
+					_timer.Start();
+				}
+
+				/// <summary>
+				/// Gets the cumulative delta, including the manipulation cumulative when this processor was started
+				/// </summary>
+				public ManipulationDelta GetCumulative()
+					=> _cumulative0.Add(_inertiaCumulative);
+
+				private void OnTick(DispatcherQueueTimer sender, object args)
+				{
+					// First we update the internal state (i.e. the current cumulative manip delta for the current time)
+					var t = _timer.LastTickElapsed.TotalMilliseconds;
+					var previous = _inertiaCumulative;
+					var current = GetInertiaCumulative(t, previous);
+
+					_inertiaCumulative = current;
+
+					// Then we request to the owner to raise its events (will cause the GetCumulative())
+					// We notify in any cases in order to make sure to raise at least one ManipDelta (even if Delta.IsEmpty ^^) before stop the processor
+					_owner.NotifyUpdate();
+
+					if (previous.Translation.X == current.Translation.X
+						&& previous.Translation.Y == current.Translation.Y
+						&& previous.Rotation == current.Rotation
+						&& previous.Expansion == current.Expansion) // Note: we DO NOT compare the scaling, expansion is enough here!
+					{
+						_timer.Stop();
+						_owner.NotifyUpdate();
+					}
+				}
+
+				private ManipulationDelta GetInertiaCumulative(double t, ManipulationDelta previousCumulative)
+				{
+					var linearX = GetValue(_isTranslateInertiaXEnabled, _velocities0.Linear.X, DesiredDisplacementDeceleration, t, (float)previousCumulative.Translation.X);
+					var linearY = GetValue(_isTranslateInertiaYEnabled, _velocities0.Linear.Y, DesiredDisplacementDeceleration, t, (float)previousCumulative.Translation.Y);
+					var angular = GetValue(_isRotateInertiaEnabled, _velocities0.Angular, DesiredRotationDeceleration, t, previousCumulative.Rotation);
+					var expansion = GetValue(_isScaleInertiaEnabled, _velocities0.Expansion, DesiredExpansionDeceleration, t, previousCumulative.Expansion);
+
+					var scale = _isScaleInertiaEnabled ? (_owner._origins.Distance + expansion) / _owner._origins.Distance : 1;
+
+					var delta = new ManipulationDelta
+					{
+						Translation = new Point(linearX, linearY),
+						Rotation = angular,
+						Expansion = expansion,
+						Scale = scale
+					};
+
+					return delta;
+				}
+
+				private float GetValue(bool enabled, double v0, double d, double t, float lastValue)
+					=> (enabled, IsCompleted(v0, d, t)) switch
+					{
+						(false, _) => 0,
+						(_, true) => lastValue, // Avoid bounce effect by replaying the last value
+						(true, false) => GetValue(v0, d, t)
+					};
+
+				// https://docs.microsoft.com/en-us/windows/win32/wintouch/inertia-mechanics#inertia-physics-overview
+				private float GetValue(double v0, double d, double t)
+					=> v0 >= 0
+						? (float)(v0 * t - d * Pow(t, 2))
+						: -(float)(-v0 * t - d * Pow(t, 2));
```

Comment:
Why are you converting to `float` here?  You'll use it in `Point()`'s constructor which is expecting `double` parameters.

Reply:
I also use it for `ManipulationDelta.Angular` and `ManipulationDelta.Rotation` which are floats. If I change it to `double` it won't reduce the number of cast but will increase memory pressure for nothing as we don't lost any significant precision here.